### PR TITLE
Persist params by selected row instead of by grapher ID

### PR DESCRIPTION
**Pull request by shaahmad**
This fixes [Changing explorer params dismisses the selected time range](https://www.notion.so/owid/Changing-explorer-params-dismisses-the-selected-time-range-8eeccfd91ba445c9a2e566e15dadf91a) and [CO2 explorer adds world and/or switches from line to map unexpectedly](https://www.notion.so/owid/CO2-explorer-adds-world-and-or-switches-from-line-to-map-unexpectedly-ae404eba49e0470bbceadb85bb2dea0c).

Basically, each grapher still gets its own query params, except the params are persisted by selected row instead of grapher ID, which works for both switcher explorers (which have a grapher ID) and non-switchers (which don't).

Certain params can be chosen to persist across explorer row changes -- currently, I'm only persisting the `time` param.

### CODE REVIEW

**Comment by shaahmad** [comment]
Deployed to Tufte (Need to brush up Nightingale .env 😄 )

**Comment by shaahmad** [comment]
Going to merge as-is since it fixes a bunch of bad bugs, but still open to further review.

**Comment by MarcelGerber** [comment]
Nice! I don't understand the code here enough that I could do a thorough review, but it looks like everything's working nicely again!
